### ISSUE TITLE
build errors on blake2bsums for files with spaces in them

### ISSUE THREAD

**Issue by christinedraper** [habitat-sh/habitat#1065]
When doing a build in studio,  there are errors generating the blake2b hashes:
```
  automateinsights: Creating manifest
   automateinsights: Building package metadata
   automateinsights: Generating blake2b hashes of all files in the package
error: File: '/hab/pkgs/christine/automateinsights/1.0.8/20160714174137/node_modules/jasmine-node/node_modules/gaze/node_modules/fileset/tests/fixtures/an' cannot be found
error: File: '/hab/pkgs/christine/automateinsights/1.0.8/20160714174137/node_modules/jasmine-node/node_modules/gaze/test/fixtures/Project' cannot be found
   automateinsights: Generating signed metadata FILES
» Signing automateinsights_blake2bsums
```
The relevant files are ones with spaces in the path, e.g.:
```
ls node_modules/jasmine-node/node_modules/gaze/test/fixtures/
nested/       one.js        Project (LO)/ sub/  
```

I think these [lines in hab-plan-build.sh](https://github.com/habitat-sh/habitat/blob/8f961a7e922b308269bb85e0dfd9440b21637b96/components/plan-build/bin/hab-plan-build.sh#L1954-1956) 

should have quotes around $file i.e.:
```
  find $pkg_prefix -type f \
    | $_sort_cmd \
    | while read file; do _b2sum "$file"; done > ${pkg_name}_blake2bsums
```

**Comment by metadave** [comment]
that fix resolves the build errors, but it doesn't look like the packaging info is quite right.

As an example, I added a file called `./config/foo bar.sh` to the redis plan. 
```
root@89f12226f7f0:/hab/pkgs/core/redis/3.0.7/20160714205429# cat FILES | grep -C 2 foo
494e16e919cfa4f3a66da7bb4fcd005fe34d69ff6c040fe93bb814409992c133  /hab/pkgs/core/redis/3.0.7/20160714205429/bin/redis-cli
020dd4d6341da80bbe96488e93f34421e619c5739e7e86004fe56f2f62678a84  /hab/pkgs/core/redis/3.0.7/20160714205429/bin/redis-server
  /hab/pkgs/core/redis/3.0.7/20160714205429/config/foo bar.sh
93c85e119f1cf2d1f4fb9a6f9b93cbd3962c377596b5adff378d755827b334a7  /hab/pkgs/core/redis/3.0.7/20160714205429/config/redis.config
e8235ccfbd1fb751bfe52ae11cb0778fd1ecfa5a255da88f8e2745b0fa95665c  /hab/pkgs/core/redis/3.0.7/20160714205429/default.toml
```

`foo bar.sh` doesn't have a blake2b hash. I'll dig into this more tomorrow AM.

Cheers-
Dave

**Comment by metadave** [comment]
I took a look this morning, but got sidetracked by some tooling issues. I'll try again early next week.

Cheers-
Dave

**Comment by metadave** [comment]
ok, I think I have a fix in PR #1076.
I built a temporary package if you'd like to try it out. You'll need to be in the Habitat dev shell:

```
make shell
export HAB_ORIGIN=foo
hab origin key generate foo
hab install metadave/hab-plan-build
$(hab pkg path metadave/hab-plan-build)/bin/hab-plan-build ./path/to/plan/dir
```

Note, that any subdirectory under `habitat` will automatically be shared to the guest OS in Docker.
I keep the `habitat-sh/core-plans` repo checked out under `habitat/plans/core-plans` and I build plans with the path `/src/plans/core-plans/zlib` etc.

Note: the metadave/hab-plan-build package is just for testing the referenced PR.
